module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -1283,18 +1281,21 @@ public function cancelHolds($cancelDetails)
             $request_json->status = 'Closed - Cancelled';
             $request_json->cancellationReasonId
                 = $this->config['Holds']['cancellation_reason'];
-            $cancel_response = $this->makeRequest(
-                'PUT',
-                '/circulation/requests/' . $requestId,
-                json_encode($request_json)
-            );
-            if ($cancel_response->getStatusCode() == 204) {
+            try {
+                $cancel_response = $this->makeRequest(
+                    'PUT',
+                    '/circulation/requests/' . $requestId,
+                    json_encode($request_json)
+                );
+                if ($cancel_response->getStatusCode() !== 204) {
```

Comment:
I don't really like this. I think we shouldn't rely on exceptions for normal operation, as it seems here. I think it would be better to just allow any status code to be returned and either mark the result success or failure based on that.